CoreUtilities/src/au/gov/asd/tac/constellation/utilities/text/SpellCheckingTextArea.java

```diff
@@ -184,47 +179,155 @@ private BooleanBinding getSelectionBinding() {
             return selectionRange == null || selectionRange.getLength() == 0;
         }, this.selectionProperty());
     }
-
-    public void autoComplete(final List<String> suggestions) {
-        final Popup popup = new Popup();
-        popup.setWidth(this.getWidth());
-        final ListView<String> listView = new ListView<>();
-        listView.getSelectionModel().selectedItemProperty().addListener((obs, oldValue, newValue) -> {
-            if (newValue != null) {
-                this.replaceText​(newValue);
+    
+     /**
+     * Moves the caret to the beginning of next word. This does not cause
+     * the selection to be cleared. Rather, the anchor stays put and the caretPosition is
+     * moved to the beginning of next word.
+     */
+    public void selectNextWord() {
+        nextWord(true);
+    }
+    
+    /**
+     * Moves the caret to the beginning of previous word. This does not cause
+     * the selection to be cleared. Rather, the anchor stays put and the caretPosition is
+     * moved to the beginning of previous word.
+     */
+    public void selectPreviousWord() {
+        previousWord(true);
+    }
+    
+    /**
+     * Moves the caret to the beginning of next word. This function
+     * also has the effect of clearing the selection.
+     */
+    public void nextWord() {
+        nextWord(false);
+    }
+  
+    /**
+     * Moves the caret to the beginning of previous word. This function also has
+     * the effect of clearing the selection.
+     */
+    public void previousWord() {
+        previousWord(false);
+    }
+    
+    /**
+     * Moves the selection forward one char in the text. This may have the
+     * effect of deselecting, depending on the location of the anchor relative
+     * to the caretPosition. This function effectively just moves the caret forward.
+     */
+    public void selectForward() {
+        final int textLength = getLength();
+        if (textLength > 0 && getCaretPosition() < textLength) {
+            if (charIterator == null) {
+                charIterator = BreakIterator.getCharacterInstance();
             }
-            popup.hide();
```

Comment:
Should add this block back.

Reply:
Verified after removing this block auto complete still works as expected for selected item from suggestions list.